posts/2024-06-07_admiral_1.1_is.../admiral_1.1_is_here.qmd

```diff
@@ -0,0 +1,248 @@
+
+# Setup
+
+The examples in this post require the following packages:
+
+```{r, message = FALSE}
+library(admiral)
+library(dplyr, warn.conflicts = FALSE)
+library(tibble)
+library(tidyverse)
+library(lubridate)
+```
+
+# Improved error messaging
+
+Our marquee update for {admiral} 1.1 is certainly our new and improved error messaging.
+Through a deep rework of our {admiraldev} package we have completely reworked how errors and warnings are thrown by {admiral} - all without creating any breaking changes for our users.
+In our new error messages:
+
+-   The name of the calling function appears on the top line to orient the user where the error is coming from.
+-   Color coding of words is now present so that the messages are easier on the eyes and easier to read.
+-   When applicable, the message is followed by an informative message (with the standard `i` prefix) providing details on what to do to address the error.
+-   Informative messages may include a hyperlink (🤯) of the code to run to get more details. Clicking the link will auto-run the code in your console and print the examples of incompatible inputs.
+
+Try it out yourself by running the code below!
+
+```{r, eval = F}
+adsl <- tribble(
+  ~USUBJID, ~TRTSDT,           ~EOSDT,
+  "01",     ymd("2020-12-06"), ymd("2021-03-06"),
+  "02",     ymd("2021-01-16"), ymd("2021-02-03")
+) %>%
+  mutate(STUDYID = "AB42")
+
+ae <- tribble(
+  ~USUBJID, ~AESTDTC,           ~AESEQ, ~AEDECOD,
+  "01",     "2021-01-03T10:56", 1,      "Flu",
+  "01",     "2021-03-04",       2,      "Cough",
+  "01",     "2021",             3,      "Flu"
+) %>%
+  mutate(STUDYID = "AB42")
+
+ae_ext <- derive_vars_dt(
+  ae,
+  dtc = AESTDTC,
+  new_vars_prefix = "AEST",
+  highest_imputation = "M",
+  flag_imputation = "none"
+)
+
+ttae <- event_source(
+  dataset_name = "ae",
+  date = AESTDT,
+  set_values_to = exprs(
+    EVNTDESC = "AE",
+    SRCDOM = "AE",
+    SRCVAR = "AESTDTC",
+    SRCSEQ = AESEQ
+  )
+)
+
+eos <- censor_source(
+  dataset_name = "adsl",
+  date = EOSDT,
+  set_values_to = exprs(
+    EVNTDESC = "END OF STUDY",
+    SRCDOM = "ADSL",
+    SRCVAR = "EOSDT"
+  )
+)
+
+derive_param_tte(
+  dataset_adsl = adsl,
+  by_vars = exprs(AEDECOD),
+  start_date = TRTSDT,
+  event_conditions = list(ttae),
+  censor_conditions = list(eos),
+  source_datasets = list(adsl = adsl, ae = ae_ext),
+  set_values_to = exprs(
+    PARAMCD = "TTAE", # This is incorrect syntax! Not unique!
+    PARAM = paste("Time to First", AEDECOD, "Adverse Event"),
+    PARCAT1 = "TTAE",
+    PARCAT2 = AEDECOD
+  )
+)
+```
+
+```{r, fig.align = 'center', echo=FALSE}
+knitr::include_graphics("admiral_error.png")
+```
+
+While we hope you don't encounter too many errors and warnings while working with {admiral}, we also hope that when you do, our improved messaging will put you on the right path towards correcting them 😀.
+
+# New metadata object for ISO 3166 dates
```

Comment:
I assume it should be "country codes" or "countries" instead of "dates".

Reply:
```suggestion
# New metadata object for ISO 3166 country codes
```